jax/_src/lax/lax.py

```diff
@@ -1138,7 +1139,7 @@ def _delta(dtype: DType, shape: Shape, axes: Sequence[int]) -> Array:
 
 def _tri(dtype: DType, shape: Shape, offset: int) -> Array:
   """Like numpy.tri, create a 2D array with ones below a diagonal."""
-  offset = int(offset)
+  assert type(offset) is int
```

Comment:
Here and elsewhere: we shouldn't assert that inputs are of type int, because (for example), this fails for numpy integer scalars.

Reply:
I add a to `int` conversion in the only caller of `_tri` -- `jax.numpy.tri`, to be consistency with  `jax.numpy.eye`.

Reply:
There's little or no downside to repeating `offset = operator.index(offset)` in place of an `isinstance` assertion; the upside is that if this low-level function is ever used elsewhere, it will be better behaved.